pkg/api/norman/customization/app/app.go

```diff
@@ -227,7 +232,7 @@ func (w Wrapper) ActionHandler(actionName string, action *types.Action, apiConte
 			return err
 		}
 
-		if err := w.validateChartCompatibility(appRevision.Status.ExternalID, clusterName); err != nil {
+		if err := w.validateChartCompatibility(appRevision.Status.ExternalID, clusterName, ""); err != nil {
```

Comment:
I'm not too familiar with this particular API, but what would be the outcome of this scenario?

1. User installs App at version x.y.z onto a cluster in Rancher 2.6.3, where this chart has a Rancher max version of 2.6.3
2. User upgrades App in-place (version remains x.y.z) with new values.yaml overrides
3. User upgrades Rancher to 2.6.4
4. User attempts to rollback values.yaml overrides

Would the rollback fail even though the version remained `x.y.z` because the Rancher version is re-evaluated? Or does a rollback strictly mean that you must go back to a previous version (e.g. `x.y.(z-1`)?

Reply:
That is already how it works today, and I think changing that would increase the surface area of this change as the issue is not concerned with rollbacks.

Reply:
I forgot that rollbacks can be to a template with the same version. I agree that this should be included and have updated.